students/brandon_nguyen/session09/mailroom_oo/donor_model.py

```diff
@@ -51,15 +68,47 @@ def __str__(self):
 
 
 class DonorCollection:
-    # Revew mailroom 1 instruction to do the following
+    """ This class will hold:
+            -all Donor objects
+            -methods to add new donor, search for a given donor
+            - extract donor
+            -create report
+            -send thank you letters to all
+    """
+    def __init__(self):
+        self.__donor_db = {}
     # We need data structure and other constants for printing
 
-    # method to update donation
-
-    # method to return donor in list getName?
+    @property
+    def donor_db(self):
+        return self.__donor_db
 
-    # method create report
+    # method to update donation
+    def add_donor_db(self, name, donation=None):
+        """ To add donor into collection"""
+        # How best to handle existing donor?
+        # TODO existing donor only take donation?
```

Comment:
You can check if the passed donor `name` already exists in your `self.__donor_db` or not, using the standard Python `key in dict` operator. Probably better check if the `name` doesn't exist in the DB and then create the entry for the new donor, and then proceed to add donation anyway:
```
if name not in self.__donor_db:
    self.__donor_db[name] = Donor(name)
self.__donor_db[name].add_donation(donation)
```